zookeeper-server/src/main/java/org/apache/zookeeper/common/StringUtils.java

```diff
@@ -48,6 +48,12 @@ private StringUtils() {/** non instantiable and non inheritable **/}
      * into a single string, where the original strings are separated using
      * the given delimiter.
      *
+     * Deprecated. Use {@link String#join(CharSequence, CharSequence...)}
```

Comment:
This should be a proper javadoc deprecated tag, something liike:

```java
/**
  * @deprecated since ZK version ... Use {@link ....} instead.`
  */
```

And, the method should have a `@Deprecated` Java annotation. Any build warnings can be suppressed by updating any calling code to suppress the warning or use its replacement.

Reply:
> Any build warnings can be suppressed by updating any calling code to suppress the warning

Yup, I was missing this piece.  Thanks!